src/core/openthread.cpp

```diff
@@ -1231,8 +1253,17 @@ void otSetReceiveIp6DatagramFilterEnabled(otInstance *aInstance, bool aEnabled)
 
 ThreadError otSendIp6Datagram(otInstance *aInstance, otMessage aMessage)
 {
-    return aInstance->mIp6.HandleDatagram(*static_cast<Message *>(aMessage), NULL, aInstance->mThreadNetif.GetInterfaceId(),
-                                          NULL, true);
+    otLogFuncEntry();
+    ThreadError error =
```

Comment:
I'd break up the variable declaration and assignment here, placing `otLogFuncEntry()` in between.